src/node/rpc/handlerregistry.h

```diff
@@ -119,6 +122,43 @@ namespace ccf
         execute_locally = v;
         return *this;
       }
+
+      Handler& restrict_allowed_verbs(std::set<http_method>&& allowed_verbs)
+      {
+        func = [prev_func = func,
+                verbs = std::move(allowed_verbs)](ccf::RequestArgs& args) {
+          const auto verb = (http_method)args.rpc_ctx->get_request_verb();
+          if (verbs.find(verb) == verbs.end())
+          {
```

Comment:
Almost certainly pointless but I wonder if this would be faster as a string lookup in a string made of all allowed verbs concatenated.

Reply:
Maybe, although we've actually got enum values rather than strings here (we can _get_ strings, but that extra step almost certainly makes it slower). Actually faster, and perhaps better, would be to build a bitmask of accepted verbs (34 verbs, comfortably in a uint64_t). With that approach we could avoid re-wrapping the func, which I'm a little uncomfortable with (it's already a deeply nested lambda), and put the "is this the correct verb" logic somewhere more obvious. I think I've talked myself into this while writing this comment, will look at doing it that way instead.

Reply:
But don't you get the enum from a map lookup/switch/equivalent on the verb string that comes in? With the string lookup you'd save that step?

Reply:
`http-parser` does this conversion for us while its parsing the request.

Reply:
The allowed verbs are now packed into a single `uint64_t`. A nice thing about this approach is that the set of verbs can be completely reset or altered by future calls to `set_allowed_verbs`, the same as any other `Handler.set_` operation (in contrast to the original lambda-wrapping approach, which could only add additional restrictions, and where repeated calls would actually produce an inaccurate `ALLOW` header response).

Reply:
Ah ok, then yeah, we might as well use a bitfield!